### PR TITLE
Delete assignment if no hours

### LINKED ISSUE #249: Assignments are deleted, rather than archived, and only under certain circumstances
Assignments can be deleted. This is an action available on the single project view or single person view. It replaces the archive button on the far right, and requires a new icon (delete rather than archive).

Assignments can be deleted if and only if there are 0 actual hours recorded (plan hours are ok). If there are actual hours recorded, the delete button is not available.

There is no concept of archiving an assignment. However, the "show/hide archived" view option should show and hide archived projects on all pages, including single project and single person views.

### PR DESCRIPTION
**Pull request by HankC138**
added ability to delete assignment if no actual hours are entered closes #249

### CODE REVIEW

File: app/components/scrollingCalendar/helpers.ts [hunk 1]
```diff
@@ -976,9 +976,10 @@ export const updateOrInsertWorkWeekInProject = (
             } else {
               updatedWorkWeeks = [...assignment.workWeeks, workWeek];
             }
+            const updatedFields = {workWeeks: updatedWorkWeeks, canBeDeleted:false};
```
Comment (zhukovdigital) [inline app/components/scrollingCalendar/helpers.ts:979]: I think this logic not working currently, notice two things: 
1. When entered estimated hours to the project/assignment - delete btn disappeared - it should disappear only if we entered actual hours I think. 
2. When delete all actual hours from project - it doesn`t appears, but should?

File: app/components/userSummary.tsx [hunk 2]
```diff
@@ -122,6 +115,24 @@ const UserSummary: React.FC<UserSummaryProps> = ({ assignment, selectedUser, pro
 		}
 	};
 
+	const handleDeleteAssignmentClick = () => {
+		if (assignment.canBeDeleted === true) {
```
Comment (zhukovdigital) [inline app/components/userSummary.tsx:119]: use just  if(assignment,caBeDeleted) instead of comparison.

File: app/components/userSummary.tsx [hunk 3]
```diff
@@ -146,14 +157,22 @@ const UserSummary: React.FC<UserSummaryProps> = ({ assignment, selectedUser, pro
 						)}
 					</div>
 				)}
-				{(viewer?.id === assignment.assignedUser?.id || !assignment.assignedUser) && <div className="sm:flex hidden items-start justify-center">
+				{(viewer?.id === assignment.assignedUser?.id && assignment.canBeDeleted === false) && <div className="sm:flex hidden items-start justify-center">
```
Comment (zhukovdigital) [inline app/components/userSummary.tsx:160]: same logic, use just !assignment.canBeDeleted

Comment (zhukovdigital) [inline app/components/userSummary.tsx:160]: Also I think we need to use same logic of showing deleting btns with viewer Id. It`s strange if other users can delete your assignment, what do u think?

Comment (HankC138) [reply app/components/userSummary.tsx:160]: 👍

File: app/components/userSummary.tsx (hunk 2 above)
Comment (HankC138) [reply app/components/userSummary.tsx:119]: 👍

File: app/components/scrollingCalendar/helpers.ts (hunk 1 above)
Comment (HankC138) [reply app/components/scrollingCalendar/helpers.ts:979]: 1. I agree
2. I believe the original intent of canBeDeleted is based on if you have workWeeks and at least in my database the workWeeks don't delete but the hours are set to zero. What is your opinion here?

File: app/components/userAssignment/workWeekInput.tsx [hunk 4]
```diff
@@ -80,6 +80,8 @@ export const WorkWeekInput = ({
 			);
 			setUserList(updatedUserList);
 			setProjectList(updatedProjectList)
+			refetchProjectList();
+			refetchUserList();
```
Comment (HankC138) [inline app/components/userAssignment/workWeekInput.tsx:84]: I don't think this is a good solution but I was unsure how to get the archive button to swap to a delete button without the refetch. Changing the field canBeDeleted is not enough for react to rerender as it is too deep in object array to notice that it has changed.

File: app/components/scrollingCalendar/helpers.ts (hunk 1 above)
Comment (zhukovdigital) [reply app/components/scrollingCalendar/helpers.ts]: > I am also noticing that when someone just erases their actualHours and submits the workWeek input empty it does not change the actualHours, is that the intended result?

good catch, I think it`s a bag. Want to create an issue? What do you about the behavior in this case? we should send 0 to delete hours?

Comment (HankC138) [reply app/components/scrollingCalendar/helpers.ts]: yes I think sending zero makes the most sense in this case

Comment (zhukovdigital) [reply app/components/scrollingCalendar/helpers.ts]: > 1. I agree
> 2. I believe the original intent of canBeDeleted is based on if you have workWeeks and at least in my database the workWeeks don't delete but the hours are set to zero. What is your opinion here?

As for canBeDeleted. I think for clear UX if assignment have no actual hours user should be able to delete this assignment as expectable behaviour. We can have work weeks only with estimated hours, but actual hours can be set to zero, Do we need additional logic for checking if some workweeks have actual hours more then 0?

File: app/components/userAssignment/workWeekInput.tsx (hunk 4 above)
Comment (HankC138) [reply app/components/userAssignment/workWeekInput.tsx]: @zhukovdigital that seems to have fixed it 👍  thank you

Comment (zhukovdigital) [reply app/components/userAssignment/workWeekInput.tsx]: @HankC138 it seems like it`s working good now, good job!

**Review by zhukovdigital — APPROVED** [review]

**Pull request merged**